### PR TITLE
[Test][e2e] Jdbc test checking data consistency. 

### PR DESCRIPTION
**Pull request by mosence**
### Purpose of this pull request

Base on PR #5560  add e2e test case.
Issues #5623 .

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Please resolve conflict

**Comment by mosence** [comment]
> Please resolve conflict

@Hisoka-X It's done. 

_Originally posted by @mosence in https://github.com/apache/seatunnel/issues/5560#issuecomment-1778965487_
`
I guess that `@TestContainers` will be mutil thread for test. so more task running on sametime. 
The result will be all task result. So in `compare` mehtod had unpredictable error occurred.
So the e2e test I will add in Issue #5623. 

Or someone can solve this problem?
image
`

**Comment by Hisoka-X** [comment]
I did some updated. Let's waiting CI.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcMysqlIT.java [hunk 1]
```diff
@@ -71,6 +77,9 @@
 import java.util.Map;
 import java.util.Properties;
 
+@DisabledOnContainer(
+        value = {},
+        type = {EngineType.FLINK, EngineType.SPARK})
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcMysqlIT.java]: Hi @mosence , this is a mistake by me. Could you revert this? Thanks.

Comment (mosence) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcMysqlIT.java]: Yes, I found it.
Why did this happen?

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcMysqlIT.java]: I do test it in local. Just want to test faster. So I did this...

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel_upper_lower.conf [hunk 2]
```diff
@@ -29,8 +29,8 @@ source {
     query = "select * from source"
     partition_column = "c_bigint_30"
     result_table_name = "jdbc"
-    partition_lower_bound = 2844674407371055160
-    partition_upper_bound = 2844674407371055259
+    partition_lower_bound = 2844674407371055000
+    partition_upper_bound = 2844674407371055099
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel_upper_lower.conf:33]: Why change this?

**Review by Hisoka-X — COMMENTED** [review]
Thanks @mosence ! Looking great now. Just one question.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel_upper_lower.conf (hunk 2 above)
Comment (mosence) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel_upper_lower.conf:33]:
```
        BigDecimal bigintValue = new BigDecimal("2844674407371055000");
        BigDecimal decimalValue = new BigDecimal("999999999999999999999999999899");
        for (int i = 0; i < 100; i++) {
            ....
            decimalValue.add(BigDecimal.valueOf(i)),
            ....
        }
```

The field preset data ranges from 2844674407371055000 to 2844674407371055099, so this configuration is necessary to integrate all data

**Review by Hisoka-X — APPROVED** [review]
Thanks @mosence !

**Review by hailin0 — APPROVED** [review]

**Pull request merged**